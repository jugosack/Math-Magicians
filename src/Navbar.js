/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
// eslint-disable-next-line import/no-extraneous-dependencies
import { Link, useMatch, useResolvedPath } from 'react-router-dom';

export default function Navbar() {
  return (
    <nav className="nav">
      <Link to="/" className="site-title">
        Math Magicians
      </Link>
      <ul>
        <CustomLink to="/">Home</CustomLink>
        <span className="divider" />
        <CustomLink to="/calculatorpage">Calculator</CustomLink>
        <span className="divider" />
        <CustomLink to="/qoutepage">Qoute</CustomLink>
      </ul>
    </nav>
  );
}

function CustomLink({ to, children, ...props }) {
  const resolvedPath = useResolvedPath(to);
  const isActive = useMatch({ path: resolvedPath.pathname, end: true });

  return (
    <li className={isActive ? 'active' : ''}>
      <Link to={to} {...props}>
        {children}
      </Link>
    </li>
  );
}
