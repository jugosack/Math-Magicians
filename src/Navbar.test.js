import React from 'react';
import { BrowserRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

const MockHeader = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

test('renders correctly', () => {
  const tree = renderer
    .create(<MockHeader />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
