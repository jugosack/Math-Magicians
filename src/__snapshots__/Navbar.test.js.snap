// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="nav"
>
  <a
    className="site-title"
    href="/"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul>
    <li
      className="active"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <span
      className="divider"
    />
    <li
      className=""
    >
      <a
        href="/calculatorpage"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <span
      className="divider"
    />
    <li
      className=""
    >
      <a
        href="/qoutepage"
        onClick={[Function]}
      >
        Qoute
      </a>
    </li>
  </ul>
</nav>
`;
